File: decentralized-api/api/model/training.go
package model

type StartTrainingDto struct {
	HardwareResources []HardwareResourcesDto `json:"hardware_resources"`
	Config            TrainingConfigDto      `json:"config"`
}

type HardwareResourcesDto struct {
	Type  string `json:"type"`
	Count uint32 `json:"count"`
}

type TrainingConfigDto struct {
	Datasets              TrainingDatasetsDto `json:"datasets"`
	NumUocEstimationSteps uint32              `json:"num_uoc_estimation_steps"`
}

type TrainingDatasetsDto struct {
	Train string `json:"train"`
	Test  string `json:"test"`
}

File: testermint/src/main/kotlin/data/training.kt
package com.productscience.data

data class StartTrainingDto(
    val hardwareResources: List<HardwareResourcesDto>,
    val config: TrainingConfigDto
)

data class HardwareResourcesDto(
    val type: String,
    val count: UInt
)

data class TrainingConfigDto(
    val datasets: TrainingDatasetsDto,
    val numUocEstimationSteps: UInt
)

data class TrainingDatasetsDto(
    val train: String,
    val test: String
)

data class LockTrainingNodesDto(
    val trainingTaskId: ULong,
    val nodeIds: List<String>
)
